src/tigerbeetle/benchmark_driver.zig

```diff
@@ -0,0 +1,126 @@
+//! Driver script behind `tigerbeetle benchmark` command.
+//!
+//! During benchmarking, there are three entities to keep track of:
+//! - the "load" process generating requests,
+//! - the cluster of `tigerbeetle`s processing requests,
+//! - the orchestrating script coordinating the two.
+//!
+//! This here is the orchestrator. If no `--addresses` is passed on the command line, it spins up a
+//! temporary single-node `tigerbeetle` cluster. Otherwise, an existing cluster is re-used for the
+//! benchmarking.
+//!
+//! The cluster address is then passed onto `benchmark_load.zig`, which deals with both offering
+//! the load and measuring response latencies and throughput. The load runs in-process.
+
+const std = @import("std");
+const assert = std.debug.assert;
+const ChildProcess = std.ChildProcess;
+
+const cli = @import("./cli.zig");
+const benchmark_load = @import("./benchmark_load.zig");
+
+const log = std.log;
+
+// Note: we intentionally don't use a temporary directory for this data file, and instead just put
+// it into CWD, as performance of TigerBeetle very much depends on a specific file system.
+const data_file = "0_0.tigerbeetle.benchmark";
```

Comment:
Could we `rm` this file if we created it when we shutdown, so we could run `tigerbeetle benchmark` multiple times without intervention?

Reply:
I guess we could actually just unlink it straight after spinning up the single node cluster? (although maybe that won't work on Windows)

Reply:
I think we do remove this file?

```
    var data_file_created = false;
    defer {
        if (data_file_created) {
            std.fs.cwd().deleteFile(data_file) catch {};
        }
    }
```

The logic here is guarded by the if because we don't want to remove the file if the user passed in the `--addresses` cli flag.

This of course doesn't work if the benchmark is killed. Unliking file immediately after format is smart, but it's not _guaranteed_ to work (we could be killed _right_ after format), and won't work on windows.

Overall, I am not yet sure what's the best way to deal with this file! I am torn between:

- just adding a random suffix to the file, so, even if the file is not deleted, it doesn't prevent concurrent benchmarks from running
- in the opposite direction, requiring the user to pass in a path to a datafile, because _where_ the data file is certainly affects performance. We, eg, don't want to put it into `/tmp`, for example.

So, I figured I preserve existing logic, and then iterate separately.

Reply:
Ah - I skipped over that! The reason I made the comment was because I was Ctrl-C'ing which I guess counts as being killed...

It'll be fairly common I think, and I don't think we need a 100% solution there - but happy to leave it up to a future iteration.

(I was partial to the random suffix in same folder idea, too)